packages/core/database/lib/entity-manager/regular-relations.js

```diff
@@ -0,0 +1,197 @@
+'use strict';
+
+const {
+  isBidirectional,
+  isOneToAny,
+  isManyToAny,
+  isAnyToOne,
+  isAnyToMany,
+} = require('../metadata/relations');
+const { createQueryBuilder } = require('../query');
+
+const getSelect = (joinTable, attribute) => {
+  const { joinColumn, orderColumnName, inverseJoinColumn, inverseOrderColumnName } = joinTable;
+  const select = [joinColumn.name, inverseJoinColumn.name];
+  if (isAnyToMany(attribute)) {
+    select.push(orderColumnName);
+  }
+  if (isBidirectional(attribute) && isManyToAny(attribute)) {
+    select.push(inverseOrderColumnName);
+  }
+  return select;
+};
+
+const deletePreviousOneToAnyRelations = async ({ id, attribute, joinTable, relIdsToadd, db }) => {
+  const { joinColumn, inverseJoinColumn, orderColumnName } = joinTable;
+  const select = getSelect(joinTable, attribute);
+
+  // need to delete the previous relations for oneToAny relations
+  if (isBidirectional(attribute) && isOneToAny(attribute)) {
+    // update orders for previous oneToAny relations that will be deleted if it has order (oneToMany)
+    if (isAnyToMany(attribute)) {
```

Comment:
I would say yes and no.
- `anyToMany`: don't have a good idea of context as we might think it's possible to have `oneToOne`. However we know we are supposed to end up in this if anyway.
- `oneToMany`: don't understand exactly why the `oneToMany` is treated but not the other relations. But we better understand what happens in the if.

In both situations, to understand what happens, we need to know that we are in a `oneToAny`. So I would say it's equivalent. I hoped the comment line 30 would be enough because I gave the information that we are in `oneToMany`.